isis/src/control/objs/ControlNet/ControlNet.cpp

```diff
@@ -817,247 +742,67 @@ namespace Isis {
-      if (uniquePoints[unprunedEdges[i]->Parent()] < 2)
-        // The point this edge is connected to does not go on to another node,
-        // so prune it
-        minimumTree.remove(unprunedEdges[i]);
-    }
-
-    // Clean up our vertices.  This will not delete any of the points, measures,
-    // or graph nodes.  All of that is owned by the network.
-    QList< ControlVertex * > vertexList = forest.values();
-    for (int i = 0; i < vertexList.size(); i++) delete vertexList[i];
-
-    // Sanity check: an island with n > 1 nodes must, by definition, have an MST
-    // of e edges such that n <= e <= 2n
-    int n = island.size();
-    int e = minimumTree.size();
-    if (n > 1) {
-      if (e < n || e > 2 * n) {
-        IString msg = "An island of n = [" + IString(n) +
-          "] > 1 nodes must have a minimum spanning tree of e edges such that "
-          " n <= e <= 2n, but e = [" + IString(e) + "]";
-        throw IException(IException::Programmer, msg, _FILEINFO_);
+    QList< QString> serials = m_vertexMap.keys(); 
+    for (int i = 0; i < serials.size(); i++) {
+      boost::put(indexMapAdaptor, m_vertexMap[serials[i]], i);
+    }
+
+    VertexIndexMap componentMap;
+    VertexIndexMapAdaptor componentAdaptor(componentMap);
+    size_t numComponents = boost::connected_components(m_controlGraph, componentAdaptor,
+                                                      boost::vertex_index_map(indexMapAdaptor));
+
+//    std::cout << "Number of components:" << std::endl;
+//    std::cout << numComponents << std::endl; 
+//    std::cout << "Size of return: " << componentMap.size() << std::endl; 
+
+    // This is not efficient, but works
+    QHash<int, QList<QString> > islandsByGroup; 
+
+    std::map<ImageVertex, size_t>::iterator it = componentMap.begin();
+    while(it != componentMap.end())
+    {
+      ImageVertex vertex = it->first;
+      QString serial = m_controlGraph[vertex].serial;
+      
+      int group = (int) it->second;
+      
+ //     std::cout << group << "  ::  "  << serial << std::endl; 
+      islandsByGroup[group].append(serial); 
+      
+      ++it;
+    }
+
+//   std::cout << std::endl; 
+
+    QList<int> groups = islandsByGroup.keys();
+    for (int i=0; i<groups.size(); i++) {
+      std::cout << "Group: " << groups[i] << std::endl; 
+      QList<QString> temp = islandsByGroup[groups[i]];
+      islandStrings.append(temp); // <--- this converts to the output format;
+      for (int j=0; j<temp.size();j++) {
+        std::cout << " ----- " << temp[j] << std::endl; 
       }
     }
 
-    return minimumTree;
+    return islandStrings;
   }
 
 
   /**
    * @returns The total number of edges in the bi-directional graph for images
    */
   int ControlNet::getEdgeCount() const {
-    int total = 0;
-    foreach (ControlCubeGraphNode * node, *cubeGraphNodes) {
-      total += node->getAdjacentNodes().size();
-    }
+    int total = boost::num_edges(m_controlGraph); 
 
     return total;
```

Comment:
smoosh to 1 line function, `return boost::num_edges(c_controlGraph)`

Reply:
:+1: